lib/spack/spack/fetch_strategy.py

```diff
@@ -322,7 +322,22 @@ def fetch(self):
 
     def _existing_url(self, url):
         tty.debug('Checking existence of {0}'.format(url))
+        # Telling urllib to check if url is accessible
         curl = self.curl
```

Comment:
If use_curl is false, we shouldn't be checking for curl - move this into the if block below.